src/recon.rs

```diff
@@ -1957,6 +1985,7 @@ pub(crate) unsafe fn rav1d_read_coef_blocks<BD: BitDepth>(
                             read_coef_tree::<BD>(
                                 f,
                                 t,
+                                Some(ts_c),
```

Comment:
It looks like we're only ever passing `Some` as I don't see any `None`.  Why does it need to be an `Option` then?

Reply:
We do pass `None` in one specific case: Towards the beginning of `decode_b`, in the `if t.frame_thread.pass == 2` block where we call `recon_b_intra` and `recon_b_inter` [we pass `None` into those functions](https://github.com/memorysafety/rav1d/blob/b41ebcad9d5505eb55c6f788be17f479aa84594f/src/decode.rs#L1227). That's the specific case where the tile state is shared between threads, but only in one branch of execution needs mutable access to the context objects.